### ISSUE TITLE
evaluate_during_training is not acceptable in newer version of the Transformer

### ISSUE THREAD

**Issue by meysamgh** [huggingface/transformers#9155]
## Environment info

     
- `transformers` version:  4.0.0
- Platform: Linux-3.10.0-1127.19.1.el7.x86_64-x86_64-with-centos-7.9.2009-Core
- Python version: 3.6.9
- PyTorch version (GPU?): 1.6.0 (True)
- Tensorflow version (GPU?): 2.3.0 (True)
- Using GPU in script?: <True>
- Using distributed or parallel set-up in script?: <False>

### Who can help
 Trainer: @sgugger

## Information

Model I am using (Bert, XLNet ...):

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [* ] my own modified scripts: (give details below)
*
config=RobertaConfig(vocab_size=30_000, max_position_embedding=512, num_attention_heads=12, num_hidden_layers=12, hidden_dropout_prob=0.1, attention_dropout_prob=0.1, initializer_range=0.2, intermediate_size=3072, type_vocab_size=1)

tokenizer=RobertaTokenizerFast.from_pretrained("XXX", max_len=512)
model=RobertaForMaskedLM(config=config)
dataset=LineByLineTextDataset(tokenizer, file_path="XXX")
data_collator=DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=True, mlm_probability=0.15)

training_args=TrainingArguments(output_dir="xxxx", overwrite_output_dir=True,
num_training_epochs=2, do_train=True, do_eval=True, **evaluate_during_training=True**,
per_gpu_train_batch_size=128, learning_rate=0.0004,
gradient_accumulation_steps=32,
logging_steps=2048,
warmup_steps=10000,
weight_decay=0.01,
eval_steps=2048,
save_steps=2048,
save_total_limit=2, prediction_loss_only=True)

trainer=Trainer(model=model, args=training_args,data_collator=data_collator, train_dataset=dataset, eval_dataset=dataseteval, **prediction_loss_only=True**)

trainer.train()

The tasks I am working on is:

* [* ] my own task or dataset: (give details below)

It is a dataset of several million lines of text.

## To reproduce

1. Having transformers installed
2. Add lines of code for accessing a training and evaluation dataset and run it

The following code was working fine with transformers 3.5 but since I updated the transformers version, it does not accept the following arguments:
evaluate_during_training=True
prediction_loss_only=True
 The error is invalid arguments for both of them. 

## Expected behavior
I need to have them to be able to check the validation loss during the model training. By removing the following arguments, the model just reports the training loss and not the evaluation loss during the training.

**Comment by sgugger** [comment]
This argument was deprecated in transformers version 3.5 and removed in version 4.0, as indicated in the [release notes](https://github.com/huggingface/transformers/releases/). It needs to be replaced by `evaluation_strategy="steps"` or `evaluation_strategy="epoch"`
